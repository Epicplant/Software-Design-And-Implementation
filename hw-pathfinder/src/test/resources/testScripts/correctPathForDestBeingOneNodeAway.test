# Tests that a dest node that is only one node away from the source node gives the correct answer

# Creates a new graph
CreateGraph newGraph

# Adds node A to the graph
AddNode newGraph A

# Adds node B to the graph
AddNode newGraph B

# Adds node C to the graph
AddNode newGraph C

# Adds edge of weight 3 to the graph
AddEdge newGraph A B 3

# Adds edge of weight 3.888 to the graph
AddEdge newGraph A C 3.888

# Adds edge of weight 2.75 to the graph
AddEdge newGraph B C 2.750

# Finds the shortest path from node A to node B
FindPath newGraph A B

# Finds the shortest path from node A to node C
FindPath newGraph A C

# Finds the shortest path from node B to node C
FindPath newGraph B C