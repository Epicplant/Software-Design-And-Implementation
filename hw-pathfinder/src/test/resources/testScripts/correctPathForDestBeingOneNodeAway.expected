# Tests that a dest node that is only one node away from the source node gives the correct answer

# Creates a new graph
created graph newGraph

# Adds node A to the graph
added node A to newGraph

# Adds node B to the graph
added node B to newGraph

# Adds node C to the graph
added node C to newGraph

# Adds edge of weight 3 to the graph
added edge 3.000 from A to B in newGraph

# Adds edge of weight 3.888 to the graph
added edge 3.888 from A to C in newGraph

# Adds edge of weight 2.75 to the graph
added edge 2.750 from B to C in newGraph

# Finds the shortest path from node A to node B
path from A to B:
A to B with weight 3.000
total cost: 3.000

# Finds the shortest path from node A to node C
path from A to C:
A to C with weight 3.888
total cost: 3.888

# Finds the shortest path from node B to node C
path from B to C:
B to C with weight 2.750
total cost: 2.750